Controller/AuthorizationController.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Trikoder\Bundle\OAuth2Bundle\Controller;
+
+use League\OAuth2\Server\AuthorizationServer;
+use League\OAuth2\Server\Exception\OAuthServerException;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
+use Symfony\Component\EventDispatcher\EventDispatcherInterface;
+use Symfony\Component\Security\Core\Security;
+use Symfony\Component\Security\Core\User\UserInterface;
+use Trikoder\Bundle\OAuth2Bundle\Event\AuthorizationRequestResolveEvent;
+use Trikoder\Bundle\OAuth2Bundle\League\Entity\User;
+use Trikoder\Bundle\OAuth2Bundle\OAuth2Events;
+use Zend\Diactoros\Response;
+
+final class AuthorizationController
+{
+    /**
+     * @var AuthorizationServer
+     */
+    private $server;
+
+    /**
+     * @var Security
+     */
+    private $security;
+
+    /**
+     * @var EventDispatcherInterface
+     */
+    private $eventDispatcher;
+
+    public function __construct(AuthorizationServer $server, Security $security, EventDispatcherInterface $eventDispatcher)
+    {
+        $this->server = $server;
+        $this->security = $security;
+        $this->eventDispatcher = $eventDispatcher;
+    }
+
+    public function indexAction(ServerRequestInterface $serverRequest): ResponseInterface
+    {
+        $serverResponse = new Response();
```

Comment:
In https://github.com/trikoder/oauth2-bundle/pull/43 we made the effort to decouple the bundle from the `Zend\Diactoros` concrete PSR-7 implementation. Please use the `Psr\Http\Message\ResponseFactoryInterface` to create the response object so that the users of the bundle can use whatever PSR-7 implementation they like.